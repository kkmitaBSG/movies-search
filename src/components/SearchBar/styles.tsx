import styled from "styled-components";
import { DebounceInput } from "react-debounce-input";

export const Input = styled(DebounceInput)`
  width: 100%;
  height: 30px;
  font-size: 20px;
  margin: 10px 0 20px;
  border-radius: 10px;
  padding-left: 10px;
`;
