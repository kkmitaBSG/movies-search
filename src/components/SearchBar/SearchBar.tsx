import React, { useCallback, useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { useSearchParams } from "react-router-dom";
import { locationSearchParameterName, MoviesActionType } from "../../store/types";
import * as S from "./styles";

export interface SearchBarProps {
  onInputChange: (searchPhrase: string) => void;
}

const SearchBar: React.FC<SearchBarProps> = ({ onInputChange }) => {
  const dispatch = useDispatch();
  const [searchParams, setSearchParams] = useSearchParams();
  const [value, setValue] = useState<string>();

  useEffect(() => {
    setValue(searchParams.get(locationSearchParameterName) || "");
  }, [searchParams]);

  const changeInput = useCallback(
    (ev: React.FormEvent<HTMLInputElement>) => {
      const target = ev.target as HTMLInputElement;
      if (target.value === "") {
        searchParams.delete(locationSearchParameterName);
        setSearchParams(searchParams);
        dispatch({ type: MoviesActionType.getMovies });

        return;
      }

      setSearchParams({ search: target.value });
      onInputChange(target.value);
    },
    [dispatch, onInputChange, searchParams, setSearchParams]
  );

  return (
    <S.InputWrapper>
      <S.BiSearchAlt />
      <S.Input
        value={value}
        placeholder={"Search a movie..."}
        minLength={1}
        debounceTimeout={500}
        onChange={changeInput}
      />
    </S.InputWrapper>
  );
};

export default SearchBar;
