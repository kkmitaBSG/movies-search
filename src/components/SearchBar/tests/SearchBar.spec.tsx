import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import * as reactRedux from "react-redux";
import SearchBar, { SearchBarProps } from "../SearchBar";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
}));

const defaultProps: SearchBarProps = {
  onInputChange: () => null,
};

describe("SearchBar", () => {
  jest.spyOn(reactRedux, "useDispatch");

  test("render", () => {
    const component = renderer.create(
      <MemoryRouter>
        <SearchBar {...defaultProps} />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
