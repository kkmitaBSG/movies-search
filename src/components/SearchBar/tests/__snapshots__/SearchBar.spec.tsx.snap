// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar render 1`] = `
<input
  className="sc-bcXHqe cEoBZN"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Search a movie..."
  type="text"
  value=""
/>
`;
